### PR TITLE
Kravetsmic/605 add include path in metadata filenam kwarg to partition functions

### PR DESCRIPTION
**Pull request by kravetsmic**
### Summary
Added the `include_path_in_metadata_filename` parameter for each partition function.
`include_path_in_metadata_filename` - determines whether or not the metadata filename will contain the full path.

If filename and metadata_filename are passed -> FILENAME_DIRECTORY/ + METADATA_FILENAME
If filename and metadata_filename are not passed -> None
If filename isn't passed and metadata_filename is passed -> METADATA_FILENAME
If filename is passed and metadata_filename isn't passed -> FILENAME

### New Parameters

`include_path_in_metadata_filename` - by default `False`

Bump to 0.8.3
-------------------------
Closes issue #605

### CODE REVIEW

File: unstructured/documents/xml.py [hunk 1]
```diff
@@ -69,6 +69,7 @@ def _read_xml(self, content):
         if self.document_tree is None:
             try:
                 document_tree = etree.fromstring(content, self.parser)
+                print("document_tree", document_tree)
```
Comment (MthwRobinson) [inline unstructured/documents/xml.py:72]: Please remove print statement

File: unstructured/documents/xml.py [hunk 2]
```diff
@@ -134,4 +142,6 @@ def from_file(
         **kwargs,
     ):
         _, content = read_txt_file(filename=filename, encoding=encoding)
+        print("_____", _)
```
Comment (MthwRobinson) [inline unstructured/documents/xml.py:145]: Please remove print statements

File: unstructured/documents/xml.py [hunk 3]
```diff
@@ -108,6 +115,7 @@ def _read_xml(self, content):
 
             self.document_tree = document_tree
 
+        print("self.document_tree", self.document_tree)
```
Comment (MthwRobinson) [inline unstructured/documents/xml.py:118]: Please remove print statement

File: unstructured/partition/common.py [hunk 4]
```diff
@@ -30,12 +31,18 @@
...
+        "LocationlessLayoutElement",
+        Element,
+        Dict[str, Any],
+    ],
     coordinate_system: Optional[CoordinateSystem] = None,
 ) -> Union[Element, List[Element]]:
     """Converts an unstructured_inference LayoutElement object to an unstructured Element."""
 
     if isinstance(layout_element, Element):
+        print("return1", layout_element.__dict__)
```
Comment (MthwRobinson) [inline unstructured/partition/common.py:45]: Please remove print statement

File: unstructured/partition/common.py [hunk 5]
```diff
@@ -52,6 +59,7 @@ def normalize_layout_element(
     # in order to add coordinates metadata to the element.
     coordinates = layout_dict.get("coordinates")
     element_type = layout_dict.get("type")
+    print("element_type", element_type)
```
Comment (MthwRobinson) [inline unstructured/partition/common.py:62]: Please remove print statement

File: unstructured/partition/html.py [hunk 6]
```diff
@@ -101,7 +102,13 @@ def partition_html(
 
         document = HTMLDocument.from_string(response.text, parser=parser)
 
-    return document_to_element_list(document, include_page_breaks=include_page_breaks)
+    print("doc", document.__dict__)
```
Comment (MthwRobinson) [inline unstructured/partition/html.py:105]: Please remove print statement

File: example-docs/html.html [hunk 7]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
```
Comment (MthwRobinson) [inline example-docs/html.html:1]: Please remove this file

File: unstructured/partition/pdf.py [hunk 8]
```diff
@@ -49,9 +49,10 @@ def partition_pdf(
     infer_table_structure: bool = False,
     ocr_languages: str = "eng",
     max_partition: Optional[int] = 1500,
+    min_partition: Optional[int] = 0,
     include_metadata: bool = True,
     metadata_filename: Optional[str] = None,
-    min_partition: Optional[int] = 0,
```
Comment (MthwRobinson) [inline unstructured/partition/pdf.py:54]: Can we keep the kwarg order the same here for backward compatibility?

File: unstructured/staging/weaviate.py [hunk 9]
```diff
@@ -60,6 +60,8 @@ def create_unstructured_weaviate_class(class_name: str = "UnstructuredDocument")
     ]
 
     for name, annotation in ElementMetadata.__annotations__.items():
+        if annotation.startswith("InitVar"):
```
Comment (MthwRobinson) [inline unstructured/staging/weaviate.py:63]: What causes the annotation to start with "InitVar"?

File: unstructured/documents/elements.py [hunk 10]
```diff
@@ -151,12 +151,13 @@ class ElementMetadata:
 
     # Metadata extracted via regex
     regex_metadata: Optional[Dict[str, List[RegexMetadata]]] = None
+    include_path_in_metadata_filename: InitVar[Optional[bool]] = None
```
Comment (MthwRobinson) [inline unstructured/documents/elements.py:154]: I'd prefer not to have this a a value in `ElementMetadata`. I don't think we'd need this if we handle setting the filename and directory name in the `process_metadata` decorator.

File: unstructured/staging/weaviate.py (hunk 9 above)
Comment (MthwRobinson) [reply unstructured/staging/weaviate.py:63]: Nevermind I see that comes from the `ElementMetadata` class above

File: unstructured/documents/elements.py (hunk 10 above)
Comment (MthwRobinson) [reply unstructured/documents/elements.py:154]: That would also eliminate the need for the `InitVar` logic you added to `weaviate.py`

**Review by MthwRobinson — CHANGES_REQUESTED** [review]
Added inline comments

File: unstructured/partition/pdf.py (hunk 8 above)
Comment (kravetsmic) [reply unstructured/partition/pdf.py:54]: Do you mean to leave `min_partition` in  54 line?

File: unstructured/documents/elements.py (hunk 10 above)
Comment (kravetsmic) [reply unstructured/documents/elements.py:154]: `InitVar` will not be presented in ElementMetadata (I mean `print(element.metadata)` - will not include with value) it is just for __post_init__ function

File: unstructured/partition/pdf.py (hunk 8 above)
Comment (kravetsmic) [reply unstructured/partition/pdf.py:54]: > Can we keep the kwarg order the same here for backward compatibility?

Do you mean to leave min_partition in 54 line?

File: unstructured/documents/elements.py (hunk 10 above)
Comment (kravetsmic) [reply unstructured/documents/elements.py:154]: > I'd prefer not to have this a a value in `ElementMetadata`. I don't think we'd need this if we handle setting the filename and directory name in the `process_metadata` decorator.

InitVar will not be presented in ElementMetadata (I mean print(element.metadata) - will not include with value) it is just for post_init function

**Comment by cragwolfe** [comment]
Closing as "won't do" for now, though this implementation was basically fine.

**Pull request closed without merging**